Feature/TA-5-On-Boarding

PeshoOReda opened the pull request:
<!--📌 Pull Request Template -->

feat(onboarding): add onboarding screen with custom buttons and text fields

- Add `EasyLocalization` package
- Implement OnBoarding screen with a logo, title, and text fields
- Create custom button and text field widgets

<!--📝 Description -->

This template ensures that all pull requests adhere to the project's contribution standards. It enforces proper naming conventions, code quality, CI/CD validation, and testing requirements. Following this checklist helps maintain a clean and efficient codebase while ensuring compatibility with JIRA issue tracking.

## 📌 Pull Request Checklist

### ✅ General
- [ ] This pull request is all my own work -- I have not plagiarized it.
- [ ] The PR title follows the required format: `type/TA-<issue-number>-TaskName`.
- [ ] The branch name follows the required format: `feature/meaningful-name`, `fix/issue-description`, etc.

### 📂 Code Quality
- [ ] All filenames are in `snake_case`.
- [ ] All functions and variable names follow Dart naming conventions.
- [ ] All new code is formatted with `dart format .`

### 🛠 CI/CD Validation
- [ ] Linting passes with `dart analyze`.
- [ ] The branch name is validated against the allowed patterns.
- [ ] The PR title is validated against the required format.

### 🚀 Testing
- [ ] Unit tests are included for all new features and bug fixes.
- [ ] `flutter test --coverage` has been run and the results reviewed.
- [ ] The project builds successfully without errors.

### 📝 Additional Information
<!-- Provide any extra context or details about the changes in this PR. -->

### 🔗 Related JIRA Ticket(s):
- [ ] JIRA Issue: `TA-<issue-number>`

### 📸 Screenshots (if applicable):
<!-- Add screenshots to show the changes visually. -->

PeshoOReda committed:
feat(onboarding): implement onboarding screen with custom buttons and text fields

PeshoOReda committed:
fix(main): update home screen to use OnBoarding as initial screen

PeshoOReda committed:
feat(devtools): add configuration file for Dart & Flutter DevTools settings

PeshoOReda committed:
no thing

PeshoOReda committed:
feat(onboarding): add onboarding screen with custom buttons and text fields

- Add `EasyLocalization` package
- Implement OnBoarding screen with a logo, title, and text fields
- Create custom button and text field widgets

PeshoOReda committed:
fix(main): Change initial screen to SplashScreen

- Updated `main.dart` to use `SplashScreen` as the initial screen instead of `OnBoarding`.

Signed-off-by: PeshoOReda <drmafiaemail13@gmail.com>

PeshoOReda committed:
refactor(main): remove unused TODO comment and unimplemented error

The unnecessary TODO comment and the UnimplementedError within the build method of _MyHomePageState have been removed as they were not being used and caused an error.

Signed-off-by: PeshoOReda <drmafiaemail13@gmail.com>

PeshoOReda committed:
Feature/TA-5-On-Boarding

ahmedelazab1220 commented on lib/features/onBoarding/presentation/view/custom_button.dart:
this is file is not accept because overwrite component in flutter make something more complex 

use ElevatedButton is more better

PeshoOReda replied:
done delete

ahmedelazab1220 commented on lib/features/onBoarding/presentation/view/on_boarding_screen.dart:
```diff
@@ -1 +1,63 @@
+              textAlign: TextAlign.start,
+            ),
+          ),
+          CustomButton(
+            label: tr(LocaleKeys.Login),
+            onPressed: () {},
+            color: AppColors.pink,
+            textStyle: const TextStyle(color: AppColors.white, fontSize: 15),
+          ),
+          CustomButton(
+            label: tr(LocaleKeys.ApplyNow),
+            onPressed: () {},
+            color: AppColors.white,
+            textStyle: const TextStyle(color: AppColors.black, fontSize: 15),
+          ),
```
this is not accept because overwrite component in flutter make something more complex

use ElevatedButton is more better

PeshoOReda replied:
doooone

ahmedelazab1220 commented on lib/features/onBoarding/presentation/view/on_boarding_screen.dart:
```diff
@@ -1 +1,63 @@
+import '../../../../core/assets/app_images.dart';
+import '../../../../core/utils/l10n/locale_keys.g.dart';
+import 'custom_button.dart';
+import 'text_field.dart' as custom;
```
not accept also because overwrite component Text is better

PeshoOReda replied:
done

ahmedelazab1220 commented on lib/main.dart:
```diff
@@ -10,17 +10,19 @@ import 'core/utils/bloc_observer/bloc_observer_service.dart';
   WidgetsFlutterBinding.ensureInitialized();
   await AppStarter.init();
-
+  await EasyLocalization.ensureInitialized();
```
this line already set in AppStarter remove that

PeshoOReda replied:
done

ahmedelazab1220 commented on lib/main.dart:
```diff
@@ -45,55 +47,18 @@ class MyApp extends StatelessWidget {
           theme: AppTheme.appTheme,
           routes: AppRoutes.routes,
-          home: const MyHomePage(title: 'Flutter Demo Home Page'),
+          home: const SplashScreen(),
```
what is ?

PeshoOReda replied:
de msh bt3ty Mstany elly hay3mlha ana mstlmo kda 😊😊😊

PeshoOReda replied:
sorry ana fakrt fi splash lsa htt3ml

ahmedelazab1220 commented on lib/features/onBoarding/presentation/view/on_boarding_screen.dart:
```diff
@@ -1 +1,63 @@
+import 'custom_button.dart';
+import 'text_field.dart' as custom;
+
+class OnBoarding extends StatelessWidget {
```
false name for class not work with standard

PeshoOReda replied:
OnBoardingScreen ?

ahmedelazab1220 commented on coverage/lcov.info:
what is?

PeshoOReda replied:
created himself 💔💔💔
i deleted it and no changes happen

PeshoOReda committed:
fix(onboarding): implement initial onboarding screen design

- Remove unused custom button and text field widgets.
- Set OnBoardingScreen as the initial screen in main.dart.
- Replace custom buttons with standard ElevatedButton.
- Adjust layout and styling for the onboarding screen.

Signed-off-by: PeshoOReda <drmafiaemail13@gmail.com>

PeshoOReda committed:
refactor(onboarding): remove unused custom button and text field files

Signed-off-by: PeshoOReda <drmafiaemail13@gmail.com>
